AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -183,6 +183,81 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
 }
 
 #pragma mark - MSDataStore Implementation
+
+- (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)__unused readOptions
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+
+  // TODO consume writeOptions
+  [MSTokenExchange
```

Comment:
Can this token acquisition step be moved into a different method or into the cosmos db class? It seems like we are doing the same thing in several places (and will need to do this in a lot of other methods in this class)

Reply:
-1 for moving token retrieval into CosmoDB, I prefer to keep them loosely coupled.

moving to a separate method won't save much, current implementation asks for token and calls completionHandler in case of error. Also, considering the Async nature of token retrieval and cosmosDB call, it requires nested calls from completion handler.

Reply:
I'm fine with them being in a separate method instead of the CosmosDB class, and I think it's worth doing. performCosmosDbAsyncOperationWithHttpClient and performDbTokenAsyncOperationWithHttpClient can be wrapped into one method, thereby reducing the nesting level, improving readability, and removing code that is or will be duplicated many times over.